Alexandr_Sytchevskiy/04.Vagrant/04.Vagrant.md

```diff
@@ -0,0 +1,89 @@
+# 04.Git.Vagrant
+
+# Links
+
+[Centos](https://app.vagrantup.com/Sanchelioss/boxes/test_centos)
+
+[Debian](https://app.vagrantup.com/Sanchelioss/boxes/test_debian)
+
+# Vagrantfile:
+
+```
+$script_centos = <<-SCRIPT
+sudo useradd -m updater
+sudo echo "updater:vagrant" | chpasswd
+sudo echo "updater ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers
+mkdir /home/updater/.ssh
+touch /home/updater/.ssh/authorized_keys
+mv /tmp/pub_key.pub /home/updater/.ssh/authorized_keys
+chown -R updater:updater /home/updater/.ssh/authorized_keys
+chmod 600 /home/updater/.ssh/authorized_keys
+sudo yum update -y
+sudo yum install curl wget git -y
+echo "System configured"
+SCRIPT
+
+$script_debian = <<-DEBSCRIPT
+export DEBIAN_FRONTEND=noniteractive
+sudo useradd -m updater
+sudo echo "updater:vagrant" | chpasswd
+sudo echo "updater ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers
+mkdir /home/updater/.ssh
+touch /home/updater/.ssh/authorized_keys
+mv /tmp/pub_key.pub /home/updater/.ssh/authorized_keys
+chown -R updater:updater /home/updater/.ssh/authorized_keys
+chmod 600 /home/updater/.ssh/authorized_keys
+sudo apt update && sudo apt upgrade -y
+sudo apt install curl wget git -y
+echo "System configured"
+DEBSCRIPT
+
+PUB_KEY_PATH = "D:/Vtemp/id_rsa.pub"
+
+MACHINES = {
+    'centos' => { vcpu: '1', ram: '1024', ip: '192.168.56.10', http_port: '8080'},
+    'debian' => { vcpu: '1', ram: '1024', ip: '192.168.56.20', http_port: '8081'},
+}
+
+INSTANCES = MACHINES.flat_map do |server_name, server|
+    server.merge({
+        :vagrant_name => "#{server_name}",
+    })
+end
+
+Vagrant.configure("2") do |config|
+  config.vm.synced_folder ".", "/vagrant_data", disabled: true
+  INSTANCES.each do |instance|
+    config.vm.define instance[:vagrant_name] do |node|
+        node.vm.box = "centos/8"
+        node.vm.hostname = "node1"
+        node.vm.network "forwarded_port", guest: 80, host: instance[:http_port]
+        node.vm.network "private_network", ip: instance[:ip]
+        if instance[:vagrant_name] .include?('deb')

```

Comment:
It is better to move this is to `MACHINES` dictionary declaration as additional parameters, like `vcpu` or `ram`